Restructure Node SDK docs. (#5710)
getsentry/sentry-docs: src/platform-includes/configuration/integrations/default-integrations/javascript.mdx

System integrations are enabled by default to integrate into the standard library or the interpreter itself. They are documented so you can be both aware of what they do and disable them if they cause issues.

## Enabled by Default

### InboundFilters

_Import name: `Sentry.Integrations.InboundFilters`_

This integration allows you to ignore specific errors based on the type,
message, or URLs in a given exception.

It ignores errors that start with `Script error` or `Javascript error: Script error` by default.

To configure this integration, use `ignoreErrors`, `denyUrls`,
and `allowUrls` SDK options directly. Keep in mind that `denyURLs` and `allowURLs`

> work only for captured exceptions, not raw message events.

Reviewer: ```suggestion only work for captured exceptions, not raw message events.
```

> work only for captured exceptions, not raw message events.

Reviewer: ```suggestion
only work for captured exceptions, not raw message events.
```

### FunctionToString

_Import name: `Sentry.Integrations.FunctionToString`_

> This integration allows the SDK to provide original functions and method names, even when our error or breadcrumbs handlers wrap them.

Reviewer: ```suggestion
This integration allows the SDK to provide original functions and method names, even when those functions or methods are wrapped by our error or breadcrumb handlers.
```

> This integration allows the SDK to provide original functions and method names, even when our error or breadcrumbs handlers wrap them.

Reviewer: ```suggestion
This integration allows the SDK to provide original functions and method names, even when those functions or methods are wrapped by our error or breadcrumb handlers.
```

Available options:

```javascript
{
  // Log calls to `console.log`, `console.debug`, etc
  console: boolean;

  // Log all click and keypress events
  // - When an object with `serializeAttribute` key is provided,
  //   Breadcrumbs integration will look for given attribute(s) in DOM elements,
  //   while generating the breadcrumb trails.
  //   Matched elements will be followed by their custom attributes,
  //   instead of their `id`s or `class` names.
  dom: boolean | { serializeAttribute: string | string[] };

  // Log HTTP requests done with the Fetch API
  fetch: boolean;

  // Log calls to `history.pushState` and friends
  history: boolean;

  // Log whenever we send an event to the server
  sentry: boolean;

  // Log HTTP requests done with the XHR API
  xhr: boolean;
}
```

### GlobalHandlers

_Import name: `Sentry.Integrations.GlobalHandlers`_

This integration attaches global handlers to capture uncaught exceptions and unhandled rejections.

Available options:

```javascript
{
  onerror: boolean;
  onunhandledrejection: boolean;
}
```

### LinkedErrors

_Import name: `Sentry.Integrations.LinkedErrors`_

> This integration allows you to configure linked errors. They’ll be recursively read up to a specified limit and lookup will be performed by a specific key. By default, the Sentry SDK sets the limit to five and the key used is `cause`.

Reviewer: ```suggestion
This integration allows you to configure linked errors. They’ll be recursively read up to a specified limit, and lookup will be performed by a specific key. By default, the Sentry SDK sets the limit to five and the key used is `cause`.
```

Available options:

```javascript
{
  key: string;
  limit: number;
}
```

Here is a code example of how this could be implemented:

```javascript
document
  .querySelector("#get-reviews-btn")
  .addEventListener("click", async event => {
    const movie = event.target.dataset.title;
    try {
      const reviews = await fetchMovieReviews(movie);
      renderMovieReviews(reviews);
    } catch (e) {
      const fetchError = new Error(`Failed to fetch reviews for: ${movie}`);
      fetchError.cause = e;
      Sentry.captureException(fetchError);
      renderMovieReviewsError(fetchError);
    }
  });
```

> ### HttpContext (previously: UserAgent)

Author: Switched to the formatting of 'New in Version ...' remarks. 👍
